pkgs/by-name/mo/modrinth-app/package.nix

```diff
@@ -0,0 +1,70 @@
+{
+  lib,
+  stdenv,
+  symlinkJoin,
+  modrinth-app-unwrapped,
+  wrapGAppsHook,
+  addOpenGLRunpath,
+  flite,
+  glib,
+  glib-networking,
+  jdk8,
+  jdk17,
+  jdks ? [
+    jdk8
+    jdk17
+  ],
+  libGL,
+  libpulseaudio,
+  udev,
+  xorg,
+}:
+symlinkJoin {
+  name = "modrinth-app-${modrinth-app-unwrapped.version}";
+
+  paths = [ modrinth-app-unwrapped ];
+
+  buildInputs = [
+    glib
+    glib-networking
+  ];
+
+  nativeBuildInputs = [ wrapGAppsHook ];
+
+  postBuild =
+    let
+      libPath = lib.makeLibraryPath [
+        flite # narrator support
+
+        udev # oshi
+
+        # lwjgl
+        libGL
+        libpulseaudio
+        stdenv.cc.cc.lib
+        xorg.libX11
+        xorg.libXcursor
+        xorg.libXext
+        xorg.libXxf86vm
+        xorg.libXrandr
+      ];
+
+      binPath = lib.makeBinPath [ xorg.xrandr ];
+
+      args =
+        [ "--prefix PATH : ${lib.makeSearchPath "bin/java" jdks}" ]
+        ++ lib.optionals stdenv.isLinux [
+          "--set LD_LIBRARY_PATH ${addOpenGLRunpath.driverLink}/lib:${libPath}"
```

Comment:
We cannot use patchelf?

Reply:
doesn't seem like it - at least in conjunction with wrapGAppsHook for manipulating PATH in this setup

Reply:
I don't think wrapGAppsHook should cause any issues here. Maybe if you run it in postFixup, you need to target the none wrapped binary instead.

Reply:
i was able to get both to apply, but it seems patching the rpath of the app itself doesn't help minecraft in detecting the libraries. setting `LD_LIBRARY_PATH` here is the only way i could get this to work